builtin/plugins/coin/coin.go

```diff
@@ -209,10 +254,223 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
+			if err != nil {
+				return errUtil.Wrap(err, "Failed Minting Block Configuration for exp operator")
+			}
+		} else {
+			amount, err = ComputeforConsecutiveYearinMiddle(ctx)
+
+		}
+	}
+	var mintingAmount = &types.BigUInt{
+		Value: *loom.NewBigUIntFromInt(0),
+	}
+	err = ctx.Get(mintingAmountKey, mintingAmount)
+	if err != nil {
+		return err
+	}
+	//Boundary case when minting amount per block becomes zero
+	if mintingAmount.Value.Cmp(loom.NewBigUIntFromInt(0)) == 0 {
+		return nil // No more coins to be minted on block creation
+	}
+	return mint(ctx, loom.UnmarshalAddressPB(policy.MintingAccount), loom.NewBigUIntFromInt(mintingAmount.Value.Int64()))
+}
+
+//First year is started from the checkpoint at which minting started
+//Computes minting Amount for first year if Blocks generated are less than a year, whether its year beginning or in the middle
+//Divides BaseAmountforfirstyear/Blocks generated per year to get minting Amount per block
+func ComputeforFirstYear(ctx contract.Context, baseAmount *common.BigUInt,
+	blocksGeneratedPerYear *common.BigUInt) (*common.BigUInt, error) {
+	amount := baseAmount.Div(baseAmount, blocksGeneratedPerYear)
+	err := ctx.Set(mintingAmountKey, &types.BigUInt{
+		Value: *amount,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return amount, nil
+}
+
+//Computes minting Amount for first year if Block Height is greater than blocks generated in a year, whether its year beginning or in the middle
+//Computes base Amount by taking minting height into Account (Height at which minting started), applies base percentage to compute inflation for that year which is divided by blocks generated per year to get ==> minting Amount per block
+func ComputeforFirstYearBlockHeightgreaterthanOneyear(ctx contract.Context, baseAmount *common.BigUInt,
+	basePercentage *common.BigUInt, blocksGeneratedPerYear *common.BigUInt) (*common.BigUInt, error) {
+	var econ Economy
+	err := ctx.Get(economyKey, &econ)
+	if err != nil {
+		return nil, err
+	}
+	//Fetches Total Supply from coin contract and applies inflation ratio on total Supply
+	baseAmount = &econ.TotalSupply.Value
+	inflationforYear := baseAmount.Mul(baseAmount, basePercentage)
+	inflationforYear = inflationforYear.Div(inflationforYear, loom.NewBigUIntFromInt(100))
+	amount := inflationforYear.Div(inflationforYear, blocksGeneratedPerYear)
+	err = ctx.Set(mintingAmountKey, &types.BigUInt{
+		Value: *amount,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return amount, nil
+}
+
+//Computes minting amount per year, after applying change ratio to base percentage using div operator
+//Computes inflation by multiplying base Amount with base percentage. Base percentage is computed by multiplying base percentage with change ratio on which div operator is applied according to the year.
+func ComputeforConsecutiveYearBeginningDivOperator(ctx contract.Context, baseAmount *common.BigUInt, changeRatioNumerator *common.BigUInt,
```

Comment:
Write a common function, which than use switch case to fan out to different operator.